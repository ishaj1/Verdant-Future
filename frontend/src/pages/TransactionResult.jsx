import { Link, useLocation } from "react-router-dom";
import useAuth from "../hooks/useAuth";

export default function TransactionResult() {
  const { auth } = useAuth();
  const { state } = useLocation();
  return (
    <div>
      <h1>Payment Successful</h1>
      <p>
        <a href={state?.details?.receipt_url} rel="noopener noreferrer" target="_blank">
          Receipt
        </a>
      </p>
      <Link to={`/profile/${auth.username}`}>Return to Profile</Link>
    </div>
  );
}
