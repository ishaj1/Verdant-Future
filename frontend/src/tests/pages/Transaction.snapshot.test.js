import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter, useLocation } from "react-router-dom";
import TransactionResult from "../../pages/TransactionResult";
import useAuth from "../../hooks/useAuth";

jest.mock("../../hooks/useAuth");
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: jest.fn(),
}));

describe("Transaction Result", () => {
  beforeEach(() => {
    useLocation.mockReturnValue({ state: { details: { receipt_url: "example.com" } } });
    useAuth.mockReturnValue({ auth: { username: "testuser" } });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("matches the snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <TransactionResult />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
