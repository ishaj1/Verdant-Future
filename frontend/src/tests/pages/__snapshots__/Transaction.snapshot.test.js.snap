// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Page matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="CreditInformation"
  >
    <div>
      Price per credit: $
      0.1
    </div>
    <div>
      Credits: 
      10
    </div>
    <div>
      Total Cost: $
      1
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    Confirm
  </button>
  <button
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;
